crates/rpc/src/method/get_transaction_status.rs

```diff
@@ -157,15 +157,26 @@ pub async fn get_transaction_status(
                 }
                 (GatewayFinalityStatus::Received, _) => Ok(Output::Received),
                 (GatewayFinalityStatus::AcceptedOnL1 | GatewayFinalityStatus::AcceptedOnL2, _) => {
-                    // This node doesn't have the accepted
-                    // transaction, meaning it's been proposed by some
-                    // other node & our view of the client's account
-                    // (in particular its nonce) is out of
-                    // date. Proposing the next transaction would fail
-                    // on the nonce mismatch. Encourage the client to
-                    // try somewhere else, hopefully where they did
-                    // last time (or later, after we update).
-                    Err(Error::TxnHashNotFound)
+                    if context
+                        .submission_tracker
+                        .contains_key(&input.transaction_hash)
+                    {
+                        // The transaction might be accepted, but it
+                        // isn't in local storage (yet). Make the
+                        // client wait some more...
+                        Ok(Output::Received)
+                    } else {
+                        // This node doesn't know anything about the
+                        // accepted transaction, meaning it's been
+                        // proposed by some other node & our view of
+                        // the client's account (in particular its
+                        // nonce) is out of date. Proposing the next
+                        // transaction would fail on the nonce
+                        // mismatch. Encourage the client to try
+                        // somewhere else, hopefully where they did
+                        // last time (or later, after we update).
+                        Err(Error::TxnHashNotFound)
+                    }
```

Comment:
Hmm, initially I thought that we need to check the submission tracker state here, but now I think we don't.

If the feeder gateway returns ACCEPTED_ON_L1 or ACCEPTED_ON_L2 we should always just return RECEIVED. In that case we can be sure that the transaction will sooner or later make it into our state, so not returning `TxnHashNotFound` is probably more correct.

```suggestion
                  // The transaction might be accepted, but it
                  // isn't in local storage (yet). Make the
                  // client wait some more...
                  Ok(Output::Received)
```